improve GCQueue comment; rename push types

tbg opened the pull request:
Fixes #3347: Recreating some transaction table entries while dealing
with stray intents and the response cache seems acceptable. If it
becomes an issue down the road, measures can be taken.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/3353)

<!-- Reviewable:end -->

tamird commented:
Reviewed 10 of 10 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[roachpb/api.proto, line 403 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3353#-K4zK9AqGyN96wWuyqmi:-K4zK9AqGyN96wWuyqmj:720900201) ([raw file](https://github.com/cockroachdb/cockroach/blob/7de5504e68073e2f6b49ecf4fb4ebcfacdef0d0e/roachpb/api.proto#L403)):*
> Inconsistent readers set this to PUSH_TOUCH to determine whether dangling intents may be resolved.
How does setting this allow this determination?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3353)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[roachpb/api.proto, line 403 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3353#-K4zK9AqGyN96wWuyqmi:-K4zQ6GoTxzn9zdqf4HR:671280421) ([raw file](https://github.com/cockroachdb/cockroach/blob/7de5504e68073e2f6b49ecf4fb4ebcfacdef0d0e/roachpb/api.proto#L403)):*
PUSH_TOUCH will abort abandoned (as per PushTxn.Now vs. txn timestamp) or nonexistent transactions only.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3353)*
<!-- Sent from Reviewable.io -->

tamird commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[roachpb/api.proto, line 403 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3353#-K4zK9AqGyN96wWuyqmi:-K51FRHOzAT2PaH8Wnsj:641454284) ([raw file](https://github.com/cockroachdb/cockroach/blob/7de5504e68073e2f6b49ecf4fb4ebcfacdef0d0e/roachpb/api.proto#L403)):*
Can you clarify that in this comment?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3353)*
<!-- Sent from Reviewable.io -->

tbg committed:
improve GCQueue comment; rename push types

Fixes #3347: Recreating some transaction table entries while dealing
with stray intents and the response cache seems acceptable. If it
becomes an issue down the road, measures can be taken.

bdarnell commented:
LGTM

tamird commented:
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3353)*
<!-- Sent from Reviewable.io -->
